python/lsst/pex/config/config.py

```diff
@@ -237,23 +237,26 @@ def save(self, outfile, instance):
         """
         value = self.__get__(instance)
         fullname = _joinNamePath(instance._name, self.name)
+
+        # write full documentation string as comment lines (i.e. first character is #)
+        doc = "# " + re.sub(r'\n', r'\n# ', str(self.doc))
         if isinstance(value, float) and (math.isinf(value) or math.isnan(value)):
             # non-finite numbers need special care
-            print >> outfile, "%s=float('%r')"%(fullname, value)
+            print >> outfile, "%s\n%s=float('%r')\n"%(doc, fullname, value)
```

Comment:
I don't think it's unreasonable to want to report something to the user from a configuration file (e.g., an optional package isn't available so I can't load configuration values from it).  Please do not outlaw print statements in configuration files without providing an alternative facility for logging problems.